sklearn/neighbors/_classification.py

```diff
@@ -154,6 +155,12 @@ class KNeighborsClassifier(KNeighborsMixin, ClassifierMixin, NeighborsBase):
     [[0.666... 0.333...]]
     """
 
+    _parameter_constraints = {**NeighborsBase._parameter_constraints}
+    _parameter_constraints.pop("radius")
+    _parameter_constraints.update(
+        {"weights": [StrOptions({"uniform", "distance"}), callable]}
```

Comment:
Looks like `None` is also valid but was not documented.
```suggestion
        {"weights": [StrOptions({"uniform", "distance"}), callable, None]}
```

Can you also update the parameter description in the docstring to reflect that None is a valid option ?